src/server_manager/web_app/survey.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {sleep} from '../infrastructure/async';
+import {Survey, Surveys} from '../model/survey';
+
+export enum SurveyId {
+  DATA_LIMITS_DISABLED = 'dataLimitsDisabledSurvey',
+  DATA_LIMITS_ENABLED = 'dataLimitsEnabledSurvey'
+}
+
+enum SurveyLink {
+  DATA_LIMITS_DISABLED =
+      'https://docs.google.com/forms/d/e/1FAIpQLSc2ZNx0C1a-alFlXLxhJ8jWk-WgcxqKilFoQ5ToI8HBOK9qRA/viewform',
+  DATA_LIMITS_ENABLED =
+      'https://docs.google.com/forms/d/e/1FAIpQLSeXQ5WUHXQHlF1Ul_ViX52GjTUPlrRB_7rhwbol3dKJfM4Kiw/viewform'
+}
+
+export const DEFAULT_PROMPT_IMPRESSION_DELAY_MS = 3000;
+
+export class OutlineSurveys implements Surveys {
```

Comment:
Ugh, I had a long comment explaining my refactor. The idea is to make the call site simpler and expose the product concepts in the model.

In model.ts:
```ts
export interface Surveys {
  presentDataLimitsEnabledSurvey();
  presentDataLimitsDisabledSurvey();
}
```

Now the model knows of all the surveys. That makes sense, since they are product concepts we expose to the users. It also simplifies the caller: no need to call requestSurvey followed by present.

The implementation will put the entire definition of a survey in one place, rather than split across different constants:
```ts
class OutlineSurveys {
  presentDataLimitsEnabledSurvey() {
    if (this.dataLimitsAvailabilityDate && now > this.dataLimitsAvailabilityDate) {
      return;
    }
    presentSurvey('dataLimitsEnabledSurvey', 'https://docs.google.com/.../viewform', DEFAULT_PROMPT_IMPRESSION_DELAY_MS)
  }
}
```

This will make the logic for each survey easier to read and understand.

No need to have SurveyIds

Reply:
Thanks for the suggestion. Done in b2fb773.